Sources/ScriptParse/ScriptParse.swift

```diff
@@ -0,0 +1,105 @@
+        let syntaxTree = try SyntaxParser.parse(path.asURL)
+
+        var inPackageScope = false
+        var keyStatements: [CodeBlockItemSyntax] = []
+        for statement in syntaxTree.statements {
+            if statement.statementKind == .import && inPackageScope {
+                keyStatements.append(statement)
+            } else {
+                inPackageScope = false
+            }
+            if statement.statementKind == .package
+                && statement.nextToken?.tokenKind == .importKeyword {
+                inPackageScope = true
+                keyStatements.append(statement)
+            }
+        }
+
+        var collected: [PackageDependency] = []
+
+        try keyStatements.forEach {
+            switch $0.statementKind {
+            case .package:
+                var tokens = [TokenSyntax]($0.tokens.dropFirst(2))
+                guard tokens.first?.tokenKind == .leftParen,
+                      tokens.last?.tokenKind == .rightParen else {
+                          throw ScriptParseError.wrongSyntax
+                      }
+                tokens.removeFirst()
+                tokens.removeLast()
+                let desc = tokens.map(\.text).joined()
+                // parse the first argument
+                if let _path = try parseStringArgument(&tokens, label: "path"),
+                   case let path = AbsolutePath(_path, relativeTo: path.parentDirectory) {
+                    let name = try parseStringArgument(&tokens, label: "name")
+                    collected.append(PackageDependency(of: PackageModel(desc.replacingOccurrences(of: _path, with: path.pathString), path: path, name: name)))
+                } else if let _url = try parseStringArgument(&tokens, label: "url"),
+                          let url = URL(string: _url) {
+                    collected.append(PackageDependency(of: PackageModel(desc, url: url)))
+                }
+                // TODO: other parsing
+                else {
+                    collected.append(PackageDependency(of: PackageModel(desc)))
+                }
+
+            case .import:
+                let tokens = [TokenSyntax]($0.tokens.dropFirst())
+                guard tokens.count == 1,
+                      let moduleToken = tokens.first,
+                      case .identifier(let moduleName) = moduleToken.tokenKind
+                else { throw ScriptParseError.unsupportedSyntax }
+                var model = collected.removeLast()
+                model.modules.append(moduleName)
+                collected.append(model)
+
+            default:
+                fatalError()
+            }
+        }
+        return ScriptDependencies(sourceFile: path, modules: collected)
+    }
+
+    public func run() throws {
+        let path = try AbsolutePath(validating: file)
+        let json = try ScriptParse.manifest(for: path)
+        print(String(data: json, encoding: .utf8)!)
```

Comment:
Is this temporary for debugging, or should the JSON always be printed?  (i.e. is this what passes it back to the caller?)

Reply:
Also, you should be able to use `String(decoding: data, as: UTF8.self)` do avoid force-unwrap.  If there are invalid sequences in the byte string, it will replace them with placeholders rather than give you nothing back.

Reply:
> Is this temporary for debugging, or should the JSON always be printed? (i.e. is this what passes it back to the caller?)

This is always printed. The former design (which is still kept) is `swift-frontend -print-package-declarations <script>`, which prints the declarations in JSON and pass it to SwiftPM.